libpromises/evalfunction.c

```diff
@@ -2066,6 +2067,157 @@ static FnCallResult FnCallClassify(EvalContext *ctx,
     return FnReturnContext(is_defined);
 }
 
+/*********************************************************************/
+
+static VersionComparison GenericVersionCheck(
+        const FnCall *fp,
+        const Rlist *args)
+{
+    assert(fp != NULL);
+    assert(fp->name != NULL);
+    if (args == NULL)
+    {
+        Log(LOG_LEVEL_ERR,
+            "Policy fuction %s requires version to compare against",
+            fp->name);
+        return VERSION_ERROR;
+    }
+
+    const char *ver_string = RlistScalarValue(args);
+    VersionComparison comparison = CompareVersion(Version(), ver_string);
+    if (comparison == VERSION_ERROR)
+    {
+        Log(LOG_LEVEL_ERR,
+            "%s: Format of version comparison string '%s' is incorrect",
+            fp->name, ver_string);
+        return VERSION_ERROR;
+    }
+
+    return comparison;
+}
+
+static FnCallResult FnCallVersionMinimum(
+        ARG_UNUSED EvalContext *ctx,
+        ARG_UNUSED const Policy *policy,
+        const FnCall *fp,
+        const Rlist *args)
+{
+    const VersionComparison comparison = GenericVersionCheck(fp, args);
+    if (comparison == VERSION_ERROR)
+    {
+        return FnFailure();
+    }
+
+    return FnReturnContext((comparison == VERSION_GREATER ||
+                            comparison == VERSION_EQUAL));
```

Comment:
Why the double parens? You could group the 2 comparisons, if you want, but having double parens at the outer level just looks weird.